numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
I would suggest using the `_wrapit` functionality here, as used elsewhere in this file. `_wrapit` uses duck typing, similarly to @mhvk's suggestion.